apps/test-low-memory.sh

```diff
@@ -1,50 +1,55 @@
 #!/bin/bash
 set -e
 
-NPROC=$(nproc)
 MEM_PER_PROCESS=4096
-GRUNT_CMD="node --max_old_space_size=${MEM_PER_PROCESS} `npm bin`/grunt"
+
+function linuxNumProcs() {
+  local nprocs=$(nproc)
+
+  # Use MemAvailable when available, otherwise fall back to MemFree
+  if grep -q MemAvailable /proc/meminfo; then
+    local mem_metric=MemAvailable
+  else
+    local mem_metric=MemFree
+  fi
+
+  # Don't run more processes than can fit in free memory.
+  local mem_procs=$(awk "/${mem_metric}/ {printf \"%d\", \$2/1024/${MEM_PER_PROCESS}}" /proc/meminfo)
+  local procs=$(( ${mem_procs} < ${nprocs} ? ${mem_procs} : ${nprocs} ))
+
+  if [ "$PROCS" -eq 0 ]; then
```

Comment:
I see this output when running the script:
```
./test-low-memory.sh: line 20: [: : integer expression expected
```
...although it didn't stop the script proceeding with parallelism=2.  Also `PROCS` isn't defined yet. My shell is rusty, but I think this works:

```suggestion
  if ((procs == 0)); then
```

Reply:
Tested and it does work. thanks!